client/clientutil/snapinfo.go

```diff
@@ -142,55 +133,13 @@ func ClientAppInfosFromSnapAppInfos(apps []*snap.AppInfo) ([]client.AppInfo, err
 		}
 
 		appInfo.Daemon = app.Daemon
-		if !app.IsService() || !app.Snap.IsActive() {
+		if !app.IsService() || decorator == nil || !app.Snap.IsActive() {
```

Comment:
It was a bit unexpected to me that this decorator is only run for active snaps. It makes sense of course but I wonder if the decision for ignoring inactive snaps should happen inside the decorator? I.e. I wonder if clientutil should know about active/inactive (or even service or not) or if that's knowledge that should be art of the decorator. This way the unit test below would also be slightly easy to read as the decorator code would run for active/non-active in the same way.

Reply:
I agree it would make the code a bit simpler here but OTOH apps don't have a status, so why call a status decorator on something that doesn't have a status? Maybe the doc comment of the functions should make that clearer.

Reply:
Yeah, maybe a small doc comment. Or renaming to "AppDecorator" and running for all apps (active or not) but either way totally fine :)